finality-aleph/src/new_network/connection_manager/connections.rs

```diff
@@ -0,0 +1,85 @@
+use crate::{new_network::PeerId, SessionId};
+use std::collections::{HashMap, HashSet};
+
+/// Keeps track of connections we should maintain taking into account data from many sessions.
+pub struct Connections {
+    associated_sessions: HashMap<PeerId, HashSet<SessionId>>,
+    peers_by_session: HashMap<SessionId, HashSet<PeerId>>,
+}
+
+impl Connections {
+    /// Creates a new object, initially without any connections.
+    pub fn new() -> Self {
+        Connections {
+            associated_sessions: HashMap::new(),
+            peers_by_session: HashMap::new(),
+        }
+    }
+
+    /// Mark the specified peers as ones we should be connected to for the given session.
+    pub fn add_peers(&mut self, session_id: SessionId, peers: HashSet<PeerId>) {
```

Comment:
```suggestion
         pub fn add_peers(&mut self, session_id: SessionId, peers: impl Iterator<Item = PeerId>) {
```

Reply:
Done.

Reply:
Although I did `IntoIterator` rather than `Iterator`.

Reply:
That's even better.